### ISSUE TITLE
Unsupported parameters for (setup) module: gather_subset, gather_timeout

### ISSUE THREAD

**Issue by lhanson** [opoplawski/ansible-pfsense#44]
I'm trying to get basic setup functionality working for the following play:

```
- name: Configure pfSense
  hosts: pfsense
  remote_user: 'root'
  collections:
    - pfsensible.core
  tasks:
    - name: General setup
      pfsensible.core.setup:
        timezone: America/Chicago
```
I've installed the collection via `ansible-galaxy collection install pfsensible.core -p ./collections` and have set `collections_paths = collections` in `ansible.cfg`.

The setup module fails with the following error:
```
$ ansible-playbook --inventory hosts pfsense.yml

PLAY [Configure pfSense] ******************************************************************************************

TASK [Gathering Facts] ********************************************************************************************
fatal: [pfsense]: FAILED! => changed=false 
  ansible_facts: {}
  failed_modules:
    setup:
      ansible_facts:
        discovered_interpreter_python: /usr/local/bin/python2.7
      failed: true
      invocation:
        module_args:
          gather_subset:
          - all
          gather_timeout: 10
      msg: 'Unsupported parameters for (setup) module: gather_subset, gather_timeout Supported parameters include: dashboardavailablewidgetspanel, dashboardcolumns, disablealiaspopupdetail, dns_addresses, dns_gateways, dns_hostnames, dnsallowoverride, dnslocalhost, domain, hostname, interfacessort, language, logincss, loginshowhost, requirestatefilter, roworderdragging, statusmonitoringsettingspanel, systemlogsfilterpanel, systemlogsmanagelogpanel, timeservers, timezone, webguicss, webguifixedmenu, webguihostnamemenu, webguileftcolumnhyper'
  msg: |-
    The following modules failed to execute: setup

PLAY RECAP ********************************************************************************************************
pfsense                    : ok=0    changed=0    unreachable=0    failed=1    skipped=0    rescued=0    ignored=0
```
Version information from the pfSense web UI:
```
2.4.4-RELEASE-p3 (amd64)
built on Wed May 15 18:53:44 EDT 2019
FreeBSD 11.2-RELEASE-p10

The system is on the latest version.
Version information updated at Mon Mar 23 19:05:18 UTC 2020
```

**Comment by lhanson** [comment]
After some googling, I discovered that `gather_subset` and `gather_timeout` are part of the built-in setup module (https://docs.ansible.com/ansible/latest/modules/setup_module.html) automatically called by playbooks. Sensing some namespace clashing here, I removed the `collections:` block from the play, and this worked:

```
- name: Configure pfSense
  hosts: pfsense
  remote_user: 'root'
  tasks:
    - name: General setup
      pfsensible.core.setup:
        timezone: America/Chicago
```
I was basing my usage on the example and subsequent comments explaining the new task name changes in #39. So perhaps some documentation clarity would help others avoid the same trap?

**Comment by opoplawski** [comment]
Using `collections:` in a playbook just seems like a recipe for disaster.  I'm not sure how one is expected to avoid namespace collisions using that.

**Comment by lhanson** [comment]
This is my first encounter with collections, so I was just trying to piece together how it's supposed to work. The ansible documentation [specifically mentions](https://docs.ansible.com/ansible/latest/user_guide/collections_using.html#using-collections-in-a-playbook) using collections in playbooks, so it didn't seem that far-fetched since it wasn't working to begin with because ansible had no idea what I meant by `pfsense_setup`, and I didn't know the examples in the documentation were wrong.

**Comment by opoplawski** [comment]
Part of the issue is that we have two versions of "ansible-pfsense" - this repository which was organized around eventual merging into ansible proper, and "pfsensible{.core}" which is structured like a collection.  pfsensible is currently built by a script using ansible-pfsense as the source.  There is no separate wiki docs for pfsensible yet.

**Comment by lhanson** [comment]
Ok, thanks for the info. I'll be watching with interest as I'm interested in ansible-izing as much of my pfSense install as I can, and will try to help out if I can.

**Comment by opoplawski** [comment]
I've filed https://github.com/ansible/ansible/issues/69195 to try to get some better clarification from ansible about module name collisions.  However, I think we're going to want to switch the pfsensible.core module names back to the pfsense_ prefixed versions to avoid these collisions.  I'm not sure that adding collections to the namespace search was an option when we first started on the pfsensible.core collection and I was wanting to avoid long module names.  Having the search path avoids that.

We would need to announce the change well though as it will break existing usage, but now may be the time to do it.
